sherpa/models/model.py

```diff
@@ -440,10 +440,9 @@ def cache_model(cls, pars, xlo, *args, **kwargs):
             integrate = kwargs.get('integrate', False)
 
         data = [np.array(pars).tobytes(),
-                boolean_to_byte(integrate),
-                np.asarray(xlo).tobytes()]
-        if args:
-            data.append(np.asarray(args[0]).tobytes())
+                boolean_to_byte(integrate)]
```

Comment:
Thinking out loud: so, this will invalidate the cache contents if a user tries to load in a save/pickled file (since the "schema" used to calculate the digest/hash has changed).

I am okay with this. There's also the possibility that we don't explicitly restore the cache from picked files (I seem to remember @hamogu talking about something like this with some recent changes, but I forget which); I also don't think we need to do anything to change the current system here.

Is it worth documenting somewhere - e.g. a generic "Cache results are not guaranteed to remain consistent if models are restored from a pickled file." type of statement? I don't think so, but thought I'd raise it.